inference-engine/tests/functional/plugin/shared/src/single_layer_tests/gather_nd.cpp

```diff
@@ -0,0 +1,105 @@
+    LayerTestsUtils::TargetDevice targetDevice;
+    std::map<std::string, std::string> config;
+
+    std::tie(params, netPrc, indicesPrc, inLayout, outLayout, targetDevice, config) = obj.param;
+
+    std::size_t batchDims;
+    IE::SizeVector inputShape, indicesShape;
+    std::tie(batchDims, inputShape, indicesShape) = params;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShape) << "_";
+    result << "indicesShape=" << CommonTestUtils::vec2str(indicesShape) << "_";
+    result << "batchDims=" << batchDims << "_";
+    result << "inPRC=" << netPrc.name() << "_";
+    result << "indicesPRC=" << indicesPrc.name() << "_";
+    result << "outPRC=" << netPrc.name() << "_";
+    result << "inL=" << inLayout << "_";
+    result << "outL=" << outLayout << "_";
+    result << "trgDev=" << targetDevice << "_";
+    return result.str();
+}
+
+void GatherNDLayerTest::SetUp() {
+    gatherNDLayerParams layerParams;
+    IE::Precision netPrc, indicesPrc;
+    std::map<std::string, std::string> config;
+
+    std::tie(layerParams, netPrc, indicesPrc, inLayout, outLayout, targetDevice, config) =
+        this->GetParam();
+
+    // FIXME Those precisions are used as actual network/layer data types
+    // Can't change it to be FP16
+    inPrc = InferenceEngine::Precision::FP32;
+    outPrc = InferenceEngine::Precision::FP32;
+
+    configuration.insert(begin(config), end(config));
+
+    // Unpack layer params
+    std::size_t batchDims;
+    IE::SizeVector inputShape, indicesShape;
+    std::tie(batchDims, inputShape, indicesShape) = layerParams;
+
+    auto gpInPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrc);
+    auto params = ngraph::builder::makeParams(gpInPrc, {inputShape});
+    auto paramsOuts = ngraph::helpers::convert2OutputVector(
+        ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+    // Generate indices values
+    const auto indices = [&] {
+        const auto indicesCount = std::accumulate(
+            begin(indicesShape), prev(end(indicesShape)), 1ull, std::multiplies<std::size_t>{});
+        const auto indexSize = indicesShape.back();
+
+        const auto maxDim = *std::max_element(begin(inputShape), end(inputShape));
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::uniform_int_distribution<int> distrib(1, maxDim);
+
+        std::vector<int> indicesValues;
+        indicesValues.reserve(indicesCount * indexSize);
+        for (int i = 0; i < indicesCount; i++) {
+            for (int dim = 0; dim < indexSize; dim++) {
+                indicesValues.push_back(distrib(gen) % inputShape[dim]);
+            }
+        }
+
+        const auto gpIndicesPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(indicesPrc);
+
+        return ngraph::opset5::Constant::create(gpIndicesPrc, indicesShape, indicesValues);
+    }();
+
+    auto gatherND = std::make_shared<ngraph::opset5::GatherND>(paramsOuts[0], indices, batchDims);
```

Comment:
Could you move `gatherND` creation logic to the builders? (lambda & gathernd creation)

Reply:
Done